COMETwebapp/Pages/ModelDashboard/DataDetails.razor

```diff
@@ -96,32 +97,50 @@ Copyright (c) 2022 RHEA System S.A.
     /// </summary>
     public ObservableCollection<BuiltParameter> Parameters = new ObservableCollection<BuiltParameter>();
 
+    /// <summary>
+    /// Listeners for the components to update it with ISession
+    /// </summary>
+    private Dictionary<string, IDisposable> Listeners = new Dictionary<string, IDisposable>();
+
     /// <summary>
     /// Initialize component at first render and after session update
     /// </summary>
     protected override void OnInitialized()
     {  
         this.InitializeParameters();
 
-        CDPMessageBus.Current.Listen<SessionEvent>().Where(x => x.Status == SessionStatus.EndUpdate).Subscribe(x => {
-            this.InitializeParameters();
-            this.StateHasChanged();
-        });
+        if (!Listeners.Keys.Contains("SessionEvent"))
+        {
+            Listeners.Add("SessionEvent", CDPMessageBus.Current.Listen<SessionEvent>().Where(x => x.Status == SessionStatus.EndUpdate).Subscribe(x => {
+                this.InitializeParameters();
+                this.StateHasChanged();
+            }));
+        }
+    }
+
+    /// <summary>
+    /// Stop and clear listeners of the component
+    /// </summary>
+    public void Dispose()
+    {
+        Listeners.Values.ToList().ForEach(l => l.Dispose());
+        Listeners.Clear();
     }
 
     /// <summary>
     /// Initialize the <see cref="BuiltParameter"> list showed in the table
     /// </summary>
     private void InitializeParameters()
     {
-        if (Criteria is not null && Domain is not null && ISessionAnchor.OpenIteration != null)
+        var iteration = ISessionAnchor.GetIteration();
+        if (Criteria is not null && Domain is not null && iteration != null)
```

Comment:
please prefix the local properties with the `this` keyword, e.g. `this.Criteria`